runtime/lua/vim/keymap.lua

```diff
@@ -135,4 +135,112 @@ function keymap.del(modes, lhs, opts)
   end
 end
 
+--- @class vim.keymap.get.Opts
+--- @inlinedoc
+---
+--- Lhs of mapping
+--- @field lhs? string
+---
+--- Patter to match against rhs of mapping
+--- @field rhs? string
+---
+--- Get a mapping for a certain buffer
+--- @field buffer? integer|boolean
+
+--- @class vim.keymap.get.Return.Opts
+--- @inlinedoc
+---
+--- If they mapping is an expr mapping
+--- @field expr boolean
+---
+--- If they mapping is a buffer-local mapping
+--- @field buffer boolean
+---
+--- Description of the mapping
+--- @field desc? string
+
+--- @class vim.keymap.get.Return
+--- @inlinedoc
+---
+--- Lhs of mapping
+--- @field lhs string
+---
+--- Rhs of mapping (can be callback)
+--- @field rhs string|function
+---
+--- Get a mapping for a certain buffer
+--- @field opts? integer|boolean
+
+--- Gets mappings in a format easily usable for vim.keymap.set
+--- Examples:
+---
+--- ```lua
+--- -- Gets all normal mode keymaps
+--- vim.keymap.get('n')
+---
+--- -- Gets a mapping which maps to a certain rhs
+--- vim.keymap.get('n', { rhs = "<Plug>(MyAmazingFunction)" })
+--- ```
+---
+---@param modes string|string[]
+---@param opts? vim.keymap.get.Opts
+---@return vim.keymap.get.Return[]
+function keymap.get(modes, opts)
```

Comment:
is `opts` a filter here? I think we prefer to call it such if so. Example: https://github.com/neovim/neovim/blob/069451bb214bd9d97273ac92b37a25054df0f1a8/runtime/lua/vim/lsp.lua#L773-L775

```suggestion
---@param opts? vim.keymap.get.Filter
---@return vim.keymap.get.Return[]
function keymap.get(modes, filter)
```